src/vm/coreassemblyspec.cpp

```diff
@@ -183,23 +183,35 @@ VOID  AssemblySpec::Bind(AppDomain      *pAppDomain,
                           &pPrivAsm);
     }
 
+    ICLRPrivAssembly *pBoundAssembly = NULL;
+
     if(SUCCEEDED(hr))
     {
         _ASSERTE(pPrivAsm != nullptr);
-        result = BINDER_SPACE::GetAssemblyFromPrivAssemblyFast(pPrivAsm.Extract());
+        pBoundAssembly = pPrivAsm.Extract();
+        result = BINDER_SPACE::GetAssemblyFromPrivAssemblyFast(pBoundAssembly);
         _ASSERTE(result != nullptr);
     }
     
     pResult->SetHRBindResult(hr);
     if (SUCCEEDED(hr))
     {
-        BOOL fIsInGAC = pAppDomain->IsImageFromTrustedPath(result->GetNativeOrILPEImage());
+        BOOL fIsInGAC = FALSE;
         BOOL fIsOnTpaList = FALSE;
-        const SString &sImagePath = result->GetNativeOrILPEImage()->GetPath();
-        if (pTPABinder->IsInTpaList(sImagePath))
+
+        _ASSERTE(pBoundAssembly != NULL);
+        
+        // Only initialize TPA/GAC status if we bound using DefaultContext
+        if (AreSameBinderInstance(pTPABinder, reinterpret_cast<ICLRPrivBinder *>(pBoundAssembly)))
         {
-            fIsOnTpaList = TRUE;
+            pAppDomain->IsImageFromTrustedPath(result->GetNativeOrILPEImage());
```

Comment:
I looked at both this call and the next one. The only difference is that IsImageFromTrustedPath, for sandboxed environments, would only return true if image is on TPA.

Given that we dont intend to support sandboxed environments in .NET Core, I am fine with removing this call. BTW, the value of the above call needs to be assigned to fIsInGAC - I will submit a fix it based upon whether we want to keep it or not (from the sandboxing standpoint).